packages/teko/src/Teko_Utilities.cpp

```diff
@@ -234,21 +234,21 @@ RCP<Tpetra::CrsMatrix<ST,LO,GO,NT> > buildGraphLaplacian(int dim,ST * coords,con
                                                        stencil.getGlobalMaxNumRowEntries()+1));
 
    // allocate an additional value for the diagonal, if neccessary
-   std::vector<ST> rowData(stencil.getGlobalMaxNumRowEntries()+1);
-   std::vector<GO> rowInd(stencil.getGlobalMaxNumRowEntries()+1);
+   auto sz = stencil.getGlobalMaxNumRowEntries()+1; 
+   Teuchos::Array<ST> rowData(sz);
+   Teuchos::Array<GO> rowInd(sz);
 
    // loop over all the rows
    for(LO j=0;j< (LO) gl->getNodeNumRows();j++) {
       GO row = gl->getRowMap()->getGlobalElement(j);
       ST diagValue = 0.0;
-      GO diagInd = -1;
+      GO diagInd = Teuchos::OrdinalTraits<GO>::invalid();
       size_t rowSz = 0;
 
       // extract a copy of this row...put it in rowData, rowIndicies
       stencil.getGlobalRowCopy(row,Teuchos::ArrayView<GO>(rowInd),Teuchos::ArrayView<ST>(rowData),rowSz);
- 
       // loop over elements of row
-      for(size_t i=0;i<rowSz;i++) {
+      for(GO i=0;i<rowSz;i++) {
```

Comment:
I forgot to mention that `rowSz` is a `size_t` here, so this may cause signed/unsigned comparison warnings.

Reply:
Good call... I don't know why I made this change now that I look at it.  I just changed it back.